/*!
* @file
* @brief IQuantumPath
*
* @copyright Copyright (C) University of Bristol 2017
*    This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. 
*    If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
*    See LICENSE file for details.
* @date 23/01/2018
* @author Richard Collins <richard.collins@bristol.ac.uk>
*/
syntax = "proto3";
import public "google/protobuf/empty.proto";
import public "Site.proto";

package cqp.remote;

/**
 * @brief The IPrivacyAmplify class
 */
service IQuantumPath {

    /**
    * @brief GetPath
    *   The function will calculate the best path between two points
    *   and return the result
    * @param request The two end points which are going to be used
    * @param response The calculated path, this may only contain the original hops if a direct connection can be established
    * @return status
    */
    rpc GetPath(PhysicalPathSpec) returns (PhysicalPath);

    /**
    * @brief CreatePath
    *  Construct the optical. Once this funcitons returns the path will be ready to use.
    * @param request The path to create
    * @return status
    */
    rpc CreatePath(PhysicalPath) returns (google.protobuf.Empty);
}
